service/src/main/java/bio/terra/policy/db/PaoDao.java

```diff
@@ -236,7 +231,7 @@ private void updatePao(GraphNode change) {
     PolicyInputs effectiveAttributes = change.getEffectivePolicyAttributes();
 
     // Get the dbPao and the attribute sets from the db for comparison
-    DbPao dbPao = getDbPao(pao.getObjectId());
+    DbPao dbPao = getDbPao(pao.getObjectId(), false);
```

Comment:
I'm considering setting this to `true`. This method is called by the graph walkers and while a deleted PAO is prevented from being the direct target of an update, I am not sure if we should be preventing deleted PAOs from being updated if they are a node in a graph.

Reply:
🤔 including soft deleted nodes seems like it would be desired behavior if the node is anything but a leaf node, as that is the behavior as it exists today. If this is false, it'd throw an exception for intermediate PAOs that were soft-deleted as the graph is being walked, whereas today it would include those which smells like a regression.